Xamarin.Essentials/Types/Brightness.shared.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+#if WINDOWS_UWP
+using UwpBrightness = Windows.Graphics.Display.BrightnessOverride;
+#endif
+
+namespace Xamarin.Essentials
+{
+    [Preserve(AllMembers = true)]
+    public readonly partial struct BrightnessOverride : IDisposable, IEquatable<BrightnessOverride>
+    {
+        internal BrightnessOverride(Brightness oldBrightness, Brightness appliedBrightness)
+        {
+            OldBrightness = oldBrightness;
+            AppliedBrightness = appliedBrightness;
+#if WINDOWS_UWP
+            @override = null;
+#endif
+        }
+
+#if WINDOWS_UWP
+        readonly UwpBrightness @override;
+
+        public BrightnessOverride(Brightness oldBrightness, Brightness newBrightness, UwpBrightness @override)
+            : this(oldBrightness, newBrightness)
+        {
+            this.@override = @override;
+        }
+
+        public void Dispose()
+        {
+            @override.StopOverride();
+        }
+
+        public bool Equals(BrightnessOverride other)
+        {
+            return OldBrightness.Equals(other.OldBrightness) && AppliedBrightness.Equals(other.AppliedBrightness) && @override.Equals(other.@override);
+        }
+#endif
+
+        public Brightness OldBrightness { get; }
+
+        public Brightness AppliedBrightness { get; }
+
+#if !WINDOWS_UWP
+        public void Dispose()
+        {
+            AppInfo.SetBrightness(OldBrightness);
+        }
+#endif
+
+#if !WINDOWS_UWP
+        public bool Equals(BrightnessOverride other)
+        {
+            return OldBrightness.Equals(other.OldBrightness) && AppliedBrightness.Equals(other.AppliedBrightness);
+        }
+#endif
+    }
+
+    [Preserve(AllMembers = true)]
+    public readonly struct Brightness
+    {
+        public double Value { get; }
+
+        public Brightness(double value)
+        {
+            const double minValue = 0d;
+#if __ANDROID__
+            const double minValue = -1d;
```

Comment:
It would be more convenient to use the range from 0 to 1 on Android too

Reply:
All constants must be the same across platforms. The value is actually copied to the caller, so if an assembly is compiled for one platform (including netstandard) then it will remain at the value and not actually change. Same for enum values.

Reply:
`relative_value = (value - low) / (high - low)`
`scaled_value = low_2 + (high_2 - low_2) * relative_value`
Is it possible to use this formula to convert the range for Android?
-1 on Android and 0 on iOS have same behavior?

Reply:
No they do not, which is why I have this check. Android also offers the same range as iOS for display brightness, but if you set a negative value the override ends and the is supplied brightness is chosen